spec/pr_size_checker_spec.rb

```diff
@@ -0,0 +1,194 @@
+# frozen_string_literal: true
+
+require_relative 'spec_helper'
+
+module Danger
+  describe Danger::PRSizeChecker do
+    it 'is a plugin' do
+      expect(described_class.new(nil)).to be_a Danger::Plugin
+    end
+
+    describe 'with Dangerfile' do
+      before do
+        @dangerfile = testing_dangerfile
+        @plugin = @dangerfile.pr_size_checker
+      end
+
+      context 'when checking a PR diff size' do
+        before do
+          allow(@plugin.git).to receive(:added_files).and_return([])
+          allow(@plugin.git).to receive(:modified_files).and_return([])
+          allow(@plugin.git).to receive(:deleted_files).and_return([])
+        end
+
+        shared_examples 'using the default diff size counter without a file selector' do |type|
+          let(:diff_counter_for_type) do
+            type_hash = {
+              insertions: :insertions,
+              deletions: :deletions,
+              all: :lines_of_code
+            }
+
+            type_hash[type]
+          end
+
+          it 'returns a warning when using default parameters in a PR that has larger diff than the default maximum' do
+            allow(@plugin.git).to receive(diff_counter_for_type).and_return(501)
+
+            @plugin.check_diff_size(type: type)
+
+            expect(@dangerfile.status_report[:warnings]).to eq ['Please keep the Pull Request small, breaking it down into multiple ones if necessary']
```

Comment:
I guess one should check the errors, too, for consistency with the other examples and thoroughness?

```suggestion
            expect(@dangerfile.status_report[:warnings]).to eq ['Please keep the Pull Request small, breaking it down into multiple ones if necessary']
            expect(@dangerfile.status_report[:errors]).to be_empty
```

Reply:
I think in general this isn't really a must, **except** on plugins like this one that can report either warnings or errors (which is something we might extend to all plugins). I think it makes sense to check for both 👍 I'll also add a note to myself to review all other plugins to make sure they are also following the same pattern, thanks!